### PR TITLE
Add API functions to open and resolve disputes

### PR DESCRIPTION
**Pull request by duriancrepe**
Requesting review for issue haveno-dex/haveno#161.

The test does not check for chat message via the notification system yet since that has to be implemented in issue haveno-dex/haveno#142. The dispute object does return the chat messages that are part of the dispute so that was used to verify the functionality.

### CODE REVIEW

**Comment by woodser** [comment]
Thanks @duriancrepe. Chat messages here do not depend on #142, which I figured would be complete and ready for reference by now. Since that's not done, we can implement chat functionality here first. It should be relatively straight forward to send dispute messages and notify when received using the notification service.

**Comment by duriancrepe** [comment]
Hi @woodser, I'll get that done and rebase.

**Comment by woodser** [comment]
Thanks, I'm starting to review the changes and will submit PRs to your branches when ready.

**Comment by woodser** [comment]
> still need to correct the dispute payouts for the winner without custom amount.

Great. I will wait for this. Also, this PR needs rebased on master now that we merged a conflicting PR. I opened a PR on your branch to resolve the conflicts if it helps: https://github.com/duriancrepe/haveno-ui-poc/pull/4

**Comment by duriancrepe** [comment]
> > still need to correct the dispute payouts for the winner without custom amount.
> 
> Great. I will wait for this. Also, this PR needs rebased on master now that we merged a conflicting PR. I opened a PR on your branch to resolve the conflicts if it helps: [duriancrepe#4](https://github.com/duriancrepe/haveno-ui-poc/pull/4)

Thanks, I had to update the test and got it merged and rebased.

File: src/HavenoDaemon.test.ts [hunk 1]
```diff
@@ -922,6 +922,225 @@ test("Can complete a trade", async () => {
...
+  expect(chatNotifications.length).toBe(1);
+  expect(chatNotifications[0].getChatMessage()?.getMessage()).toEqual("Arbitrator chat message to Alice");
+  chatNotifications = getNotifications(bobNotifications, NotificationMessage.NotificationType.CHAT_MESSAGE);
+  expect(chatNotifications.length).toBe(1);
+  expect(chatNotifications[0].getChatMessage()?.getMessage()).toEqual("Arbitrator chat message to Bob");
+  
+  // arbitrator has 2 chat messages, one with attachments
+  chatNotifications = getNotifications(arbitratorNotifications, NotificationMessage.NotificationType.CHAT_MESSAGE);
+  expect(chatNotifications.length).toBe(2);
+  expect(chatNotifications[0].getChatMessage()?.getMessage()).toEqual("Bob chat message");
```
Comment (woodser) [inline src/HavenoDaemon.test.ts:1057]: Occasionally getting an error on this line that the first notification is Alice's chat message. I think we just need to add `await wait(1000)` after line 1024 so Bob's chat message is sent with enough time always be first.

File: src/HavenoDaemon.test.ts [hunk 2]
```diff
@@ -922,6 +922,225 @@ test("Can complete a trade", async () => {
...
+  // alice and bob reply to arbitrator chat messages
+  let attachment = new Attachment();
+  let bytes = new Uint8Array(Buffer.from("Proof Bob was scammed", "utf8"));
+  attachment.setBytes(bytes);
+  attachment.setFileName("proof.txt");
+  let attachment2 = new Attachment();
+  let bytes2 = new Uint8Array(Buffer.from("picture bytes", "utf8"));
+  attachment2.setBytes(bytes2);
+  attachment2.setFileName("proof.png");
+  await bob.sendDisputeChatMessage(bobDispute.getId(), "Bob chat message", [attachment, attachment2]);
```
Comment (woodser) [inline src/HavenoDaemon.test.ts:1024]: Add `await wait(1000);` after here per my other comment.

File: src/HavenoDaemon.test.ts [hunk 3]
```diff
@@ -922,6 +922,225 @@ test("Can complete a trade", async () => {
...
+  let bytes = new Uint8Array(Buffer.from("Proof Bob was scammed", "utf8"));
+  attachment.setBytes(bytes);
+  attachment.setFileName("proof.txt");
+  let attachment2 = new Attachment();
+  let bytes2 = new Uint8Array(Buffer.from("picture bytes", "utf8"));
+  attachment2.setBytes(bytes2);
+  attachment2.setFileName("proof.png");
+  await bob.sendDisputeChatMessage(bobDispute.getId(), "Bob chat message", [attachment, attachment2]); 
+  await alice.sendDisputeChatMessage(aliceDispute.getId(), "Alice chat message", []); 
+  await wait(TestConfig.maxTimePeerNoticeMs);
```
Comment (woodser) [inline src/HavenoDaemon.test.ts:1026]: Let's also move this wait down after line 1028 for consistency. Thanks.

File: src/HavenoDaemon.test.ts (hunk 2 above)
Comment (duriancrepe) [reply src/HavenoDaemon.test.ts]: done

File: src/HavenoDaemon.test.ts (hunk 3 above)
Comment (duriancrepe) [reply src/HavenoDaemon.test.ts]: done

File: src/HavenoDaemon.test.ts (hunk 1 above)
Comment (duriancrepe) [reply src/HavenoDaemon.test.ts:1058]: Done, I did not reproduced this so cannot verify with certainty it fixes the issue.

**Pull request merged**